lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -90,6 +107,20 @@
 	fn sign_holder_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, holder_commitment_tx: &HolderCommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()> {
 		// TODO: enforce the ChannelKeys contract - error if this commitment was already revoked
 		// TODO: need the commitment number
+		let revoked = self.revoked_commitment.lock().unwrap();
+		let keys = &holder_commitment_tx.keys;
+		if keys.per_commitment_point != self.inner.get_per_commitment_point(*revoked - 1, secp_ctx) {
+			if keys.per_commitment_point != self.inner.get_per_commitment_point(*revoked - 2, secp_ctx) {
+				if keys.per_commitment_point == self.inner.get_per_commitment_point(*revoked, secp_ctx) {
```

Comment:
I think this check can be brought out to the top. Basically, if p_c_p == current { attempted tto sign the latest } else if p_c_p != -1 && p_c_p != -2 { can only sign the next two }

Reply:
See above about performance.

Reply:
Actually, with the recent phase 2 signer API we now have the commitment number, so I modified this code to just directly compare the two numbers.  Please take a look.